imagebuildah/stage_executor.go

```diff
@@ -916,6 +929,58 @@ func (s *StageExecutor) UnrecognizedInstruction(step *imagebuilder.Step) error {
 	return errors.New(err)
 }
 
+// do our best to ensure that image specifiers that include a transport that
+// uses path names are scoped to the build context directory
```

Comment:
It'd be secure and reliable to get a directory fd for the context overlay mount, and then use `openat(..., RESOLVE_IN_ROOT)` to access the file, then...the ugly part is that the containers-storage APIs want path strings still, but we could pass e.g. `oci:/proc/self/fd/N` right? I've done that elsewhere I think.

Reply:
Proper safety for such references is a much larger patch.